### PR TITLE
Add MCP Connect button OAuth flow

### PR DESCRIPTION
**Pull request by vedworks**
## Summary
- Replaces MCP token entry during server creation with a Cursor-style add URL first, then Connect flow.
- Adds MCP server probing, state persistence, token fallback Connect, and one-click OAuth 2.1 with DCR + PKCE.
- Adds token refresh on invalid tokens, settings-row hydration probes, telemetry, and edge-case coverage for non-OAuth/DCR servers.

## Test plan
- `pnpm --dir apps/web exec vitest run lib/mcp-probe.test.ts lib/mcp-oauth.test.ts lib/mcp-servers.test.ts app/api/settings/mcp/route.test.ts app/api/settings/mcp/probe/route.test.ts app/api/settings/mcp/connect/token/route.test.ts app/api/settings/mcp/connect/start/route.test.ts app/api/settings/mcp/connect/callback/route.test.ts`
- Browser-tested Cloud settings MCP flow with a local mock MCP server: add server, Connect fallback token dialog, token submission, connected tool count, reload persistence, delete cleanup.
- Browser-tested one-click OAuth path with a local mock auth server: DCR, authorization redirect callback, token exchange, row flips to authenticated with tool count.

---

> [!NOTE]
> **High Risk**
> Adds new OAuth/token-based connection endpoints and local secret/state persistence for MCP servers, which touches authentication headers and token handling. Incorrect validation/storage/probing could break connectivity or leak tokens/state despite added tests and escaping.
> 
> **Overview**
> Adds a **Cursor-style MCP “Add then Connect” flow**: server creation no longer accepts an auth token and now immediately probes the server to populate persisted connection state.
> 
> Introduces new Connect APIs: `POST /api/settings/mcp/connect/start` to probe and (when supported) run OAuth discovery + dynamic client registration + PKCE, `GET /api/settings/mcp/connect/callback` to exchange the code, persist refresh tokens, update `Authorization` headers, probe, and `postMessage` results back to the opener, and `POST /api/settings/mcp/connect/token` as a manual token fallback that saves the Bearer header then re-probes.
> 
> Adds supporting libraries for **probing** (`tools/list` over JSON/SSE with RFC 9728 `WWW-Authenticate` parsing), **OAuth client utilities** (discovery, DCR, PKCE, code exchange, refresh, expiry computation), and **sidecar persistence** for secrets (`.mcp-secrets.json`, 0600) and per-server status (`.mcp-states.json`), including cleanup on server removal and refresh-on-`invalid_token` during probes.
> 
> Updates the settings UI to show per-server status/tool counts, add a Connect button with OAuth popup handling + timeout, fallback token dialog, and automatic hydration probes; expands test coverage across routes and new libs.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 5da08f9a15162e9d3ef57140cea6d44f5cc74e87. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: apps/web/lib/mcp-servers.ts [hunk 1]
```diff
@@ -226,4 +464,10 @@ export function removeMcpServer(key: string): void {
   }
 
   writeConfig(config);
+
+  const states = readStatesSidecar();
+  if (states[normalizedKey]) {
+    delete states[normalizedKey];
+    writeStatesSidecar(states);
+  }
```
[the remark this answers is not in the document]
Comment (vedworks) [reply apps/web/lib/mcp-servers.ts:473]: Fixed in 5da08f9a1 by calling deleteMcpServerSecret from removeMcpServer, with a regression test covering server removal cleanup.

File: apps/web/lib/mcp-secrets.ts [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+  if (!current) {
+    return;
+  }
+  all[key] = {
+    ...current,
+    codeVerifier: null,
+    oauthState: null,
+  };
+  writeAll(all);
+}
```
[the remark this answers is not in the document]
Comment (vedworks) [reply apps/web/lib/mcp-secrets.ts:191]: Fixed in 5da08f9a1 by clearing redirectUri in clearTransientOAuthFields while keeping scope for refresh-token grants, with a regression test covering the transient cleanup contract.
